certbot-apache/certbot_apache/override_centos.py

```diff
@@ -66,40 +72,50 @@ def _deploy_loadmodule_ssl_if_needed(self):
 
         loadmods = self.parser.find_dir("LoadModule", "ssl_module", exclude=False)
 
-        # We should have one in ssl.conf
-        sslconf_loadmod_path = None
-        loadmod_args = ["ssl_module", "modules/mod_ssl.so"]
-
+        # We should have at least one LoadModule ssl_module in the config
+        loadmod_args = []  # type: List[str]
+        loadmod_path = None
         for m in loadmods:
-            if "/conf.d/ssl.conf/" in m:
-                # Strip "arg[1]" off from the end
-                sslconf_loadmod_path = m[:-7]
-                # Use the preconfigured LoadModule values from ssl.conf
-                loadmod_args = self.parser.get_all_args(sslconf_loadmod_path)
-            elif self.parser.loc["default"] in m:
+            noarg_path = m.rpartition("/")[0]
+            path_args = self.parser.get_all_args(noarg_path)
+            if loadmod_args:
+                if loadmod_args != path_args:
+                    logger.info("Multiple different LoadModule directives for mod_ssl "
```

Comment:
I slightly misunderstood the changes you described on the call earlier this week. What do you think of the following pseudocode:

```
for m in loadmods:
    if m is in config root:
        return
for m in loadmods:
    if loadmod_args is different:
        raise "Multiple different LoadModule..."
add "LoadModule ssl_module" with loadmod_args to the top of the config root
```

Reply:
I'm not sure if we should raise an exception here. If the user has different `LoadModule ssl_module` directives in their configuration, I would believe they have manually tuned the configuration, and most likely at that situation know what they are doing.